pisek/task_jobs/solution/solution_manager.py

```diff
@@ -390,26 +386,34 @@ def status(
 
         raise RuntimeError(f"Unknown verbosity {verbosity}")
 
-    def _verdict_summary(self, jobs: list[RunChecker]) -> str:
-        text = ""
-        verdicts = self._verdicts(jobs)
-        for verdict in Verdict:
-            count = verdicts.count(verdict)
-            if count > 0:
-                text += f"{count}{verdict.mark()}"
-        return text
-
     def _verdict_marks(self, jobs: list[RunChecker]) -> str:
         return "".join(job.verdict_mark() for job in jobs)
 
     def _predecessor_summary(self) -> str:
-        predecessor_summary = self._verdict_summary(self.previous_jobs)
-        if predecessor_summary:
-            return f"({predecessor_summary}) "
-        return ""
+        if not self.previous_jobs:
+            return ""
+
+        verdicts = self._verdicts(self.previous_jobs)
+        if not verdicts:
+            return "p | "
+
+        _, _, guarantor = self._as_expected(self.previous_jobs)
```

Comment:
It's kind of confusing that the final return value has a different name than inside the function.